### PR TITLE
03-mongoDB

### PR DESCRIPTION
**Pull request by antuan159**

### CODE REVIEW

File: contacts.js [hunk 1]
```diff
@@ -1,117 +1,97 @@
...
     }
-    let updatecContact;
 
-    const filtredContacts = listContacts.filter(contact => {
-      if (contact.id === contactId) {
-        contact.name = name || contact.name;
-        contact.email = email || contact.email;
-        contact.phone = phone || contact.phone;
-        updatecContact = contact;
+    await Contacts.updateOne(
```
Comment (odykyi) [inline contacts.js]: read documentation 

remove callback from here

File: contacts.js [hunk 2]
```diff
@@ -1,117 +1,97 @@
...
-      return { message: 'Not found' };
+    const deleted = await getContactById(contactId);
+    if (!deleted) {
+      return { message: "object not found" };
     }
 
-    const filtredContacts = listContacts.filter(
-      contact => contact.id !== contactId
-    );
+    await Contacts.deleteOne({ _id: contactId }, function(err, cb) {
```
Comment (odykyi) [inline contacts.js]: read documentation 

remove callback from here

File: contacts.js [hunk 3]
```diff
@@ -1,117 +1,97 @@
...
-
-    if (listContacts) {
-      arr = listContacts;
-      contact['id'] = arr[arr.length - 1].id + 1;
-    }
-    arr.push(contact);
-
-    await writeFileAsync(contactPath, JSON.stringify(arr));
+    let newContact;
+    await Contacts.create(contact, function(err, contactBD) {
```
Comment (odykyi) [inline contacts.js]: read documentation 

remove callback from here

File: contacts.js [hunk 4]
```diff
@@ -1,117 +1,97 @@
...
 // this function finds contact by Id and write in console
 async function getContactById(contactId) {
   try {
-    const listContacts = await readFileAsync(contactPath, encoding);
-    const contactsList = JSON.parse(listContacts);
-
-    let obj = contactsList.filter(contact => {
-      if (contact.id === contactId) return contact;
+    let contact;
+    await Contacts.find({ _id: contactId }, function(err, contactDB) {
```
Comment (odykyi) [inline contacts.js]: read documentation 

remove callback from here

File: contacts.js [hunk 5]
```diff
@@ -1,117 +1,97 @@
...
-
+const Contacts = require("./api");
 
 // this function finds all save contact
 async function listContacts() {
   try {
-    const listContacts = await readFileAsync(contactPath, encoding);
-    return JSON.parse(listContacts);
+    let list;
+    await Contacts.find({}, function(err, listContacts) {
```
Comment (odykyi) [inline contacts.js]: read documentation 

remove callback from here

**Review by odykyi — CHANGES_REQUESTED** [review]
read documentation 

remove callbacks

**Comment by odykyi** [comment]
https://mongoosejs.com/docs/api.html

File: contacts.js (hunk 1 above)
Comment (antuan159) [reply contacts.js]: fixed

File: contacts.js (hunk 2 above)
Comment (antuan159) [reply contacts.js]: fixed

File: contacts.js (hunk 3 above)
Comment (antuan159) [reply contacts.js]: fixed

File: contacts.js (hunk 4 above)
Comment (antuan159) [reply contacts.js]: fixed

**Review by odykyi — APPROVED** [review]

**Pull request merged**